### ISSUE TITLE
HTTP request helper parses response as JSON without checking for errors

### ISSUE THREAD

**Issue by felixc** [plaid/plaid-python#106]
The `http_request` helper method in `requester.py` unconditionally attempts to parse the response as JSON: https://github.com/plaid/plaid-python/blob/02db6f522b9381361272746123ac916616d48674/plaid/requester.py#L33

By "unconditionally" I mean that it doesn't first check the HTTP response status code, nor is the `json.loads` call wrapped in a try/except block that would allow it to handle failures.

This means that if you get a response like the following, you get a `JSONDecodeError`:

```html
<html>
<head><title>502 Bad Gateway</title></head>
<body bgcolor="white">
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

Ideally, the method should either/both check the HTTP status code and/or catch exceptions from `json.loads`, and raise a custom exception as appropriate, without leaking the `JSONDecodeError`.

**Comment by joyzheng** [comment]
Thanks for reporting! Agree that we should be handling this case better -- I'll work on a PR to fix.

**Comment by michaelckelly** [comment]
Released as `2.1.1`
